### PR TITLE
1.0.0/feature/graphic 2.0

### PR DESCRIPTION
**Pull request by bapthub**
**ADD :  **
New Version of the graphic interface

**Fix**:
-`src/main.c` : changes on graphic interface
-`src/../main_window.glade` :  changes on graphic interface

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -3,34 +3,49 @@
 #include <stdio.h>
 #define UNUSED(x) (void)(x)
```
Comment (Winbay) [inline src/main.c:5]: all include ?

File: src/main.c [hunk 2]
```diff
@@ -3,34 +3,49 @@
...
     GtkBuilder      *builder;
     GtkWidget       *window;
+    GtkWidget       *button;
     app_widgets     *widgets = g_slice_new(app_widgets);
+    /*char *filename = "src/save_file/save.txt";
+    char *text_to_save = "oui";
+    UNUSED(filename);
+    UNUSED(text_to_save);*/
+    UNUSED(button);
+
```
Comment (JulesdeCube) [inline src/main.c]: to clean

File: src/main.c [hunk 3]
```diff
@@ -39,7 +54,10 @@ void on_menuitm_open_activate(GtkMenuItem *menuitem, app_widgets *app_wdgts)
 {
     UNUSED(menuitem);
     gchar *file_name = NULL;        // Name of file to open from dialog box
+    //GtkWidget *image;
     
+    //int x_pos = 0; //x and y pos of the image in the window
+    //int y_pos = 0;
```
Comment (JulesdeCube) [inline src/main.c]: to remove

File: src/main.c [hunk 4]
```diff
@@ -3,34 +3,49 @@
...
+    char      *w_path;                  //path of the image open
+    char      *w_save_path;             //path of the file where the text is save
+    char      *w_text_to_save;          //string of the text to save
 } app_widgets;
 
 int main(int argc, char *argv[])
 {
     GtkBuilder      *builder;
     GtkWidget       *window;
+    GtkWidget       *button;
```
Comment (Winbay) [inline src/main.c]: Why if u set it on "UNUSED"

File: src/main.c [hunk 5]
```diff
@@ -159,16 +200,13 @@ void on_menuitm_open_preprocess_activate(GtkMenuItem *menuitem, app_widgets *app
     gtk_widget_hide(app_wdgts->w_dlg_file_choose);
 }
 
-
-/* !  Not working !
-fonction to aplly a filter on the current image
-void on_menuitm_switch(GtkMenuItem *menuitem, app_widgets *app_wdgts)
+void on_button_clicked(GtkButton *button, app_widgets *app_wdgts)
```
Comment (JulesdeCube) [inline src/main.c]: add function comment

File: src/main.c [hunk 6]
```diff
@@ -48,16 +66,39 @@ void on_menuitm_open_activate(GtkMenuItem *menuitem, app_widgets *app_wdgts)
...
         file_name = gtk_file_chooser_get_filename(GTK_FILE_CHOOSER(app_wdgts->w_dlg_file_choose));
+        if(app_wdgts->w_path != NULL)
+            g_free(app_wdgts->w_path);
+        app_wdgts->w_path = file_name;
+    
         if (file_name != NULL) 
         {
             gtk_image_set_from_file(GTK_IMAGE(app_wdgts->w_img_main), file_name);
         }
-        g_free(file_name);
```
Comment (Winbay) [inline src/main.c:65]: need to free "file_name" ?

**Review by JulesdeCube — CHANGES_REQUESTED** [review]
need to comment function
need to remove `src/save_file/save.txt` file and add it into the `.gitignore`

**Pull request closed without merging**